chain/chain/src/chain.rs

```diff
@@ -1261,7 +1261,7 @@ impl Chain {
         on_challenge: F3,
     ) where
         F: Copy + FnMut(AcceptedBlock) -> (),
-        F2: Copy + FnMut(Vec<ShardChunkHeader>) -> (),
+        F2: Copy + FnMut((CryptoHash, Vec<ShardChunkHeader>)) -> (),
```

Comment:
Might be helpful to create a separate type for more clarity. For example it is not clear to me what this hash is